src/api/analytics/AnalyticsRequest.js

```diff
@@ -84,16 +84,25 @@ class AnalyticsRequest extends AnalyticsRequestDimensionsMixin(
         const fixedIds = Object.keys(getFixedDimensions())
 
         filters.forEach((f) => {
-            request =
-                passFilterAsDimension && fixedIds.includes(f.dimension)
-                    ? request.addDimension(
-                          f.dimension,
-                          f.items?.map((item) => item.id)
-                      )
-                    : request.addFilter(
-                          f.dimension,
-                          f.items?.map((item) => item.id)
-                      )
+            if (passFilterAsDimension && fixedIds.includes(f.dimension)) {
+                request = request.addDimension(
+                    f.dimension,
+                    f.items?.map((item) => item.id)
+                )
+            } else {
+                let filterString = f.programStage?.id
+                    ? `${f.programStage.id}.${f.dimension}`
+                    : f.dimension
+
+                if (f.filter) {
+                    filterString += `:${f.filter}`
+                }
+
+                request = request.addFilter(
+                    filterString,
+                    f.items?.map((item) => item.id)
```

Comment:
Question - I did not find any examples of the items array being populated. Is it in use for this particular case?

Reply:
Items are only for "regular dimensions" (i.e. that don't have conditions), like org unit and periods. The ones that use conditions (most) won't have items.